README.md

```diff
@@ -259,37 +366,40 @@ Let's fire up the OCaml top-level:
 
 ```ocaml
 $ ocaml
-        OCaml version 4.02.2+multicore-dev0
+OCaml version 5.0.0~beta1
 
-# effect E : unit;; 
-effect E : unit  (* E is a nullary effect that returns unit *)
+# open Effect;;
+# type _ Effect.t += E : unit Effect.t;;
+type _ Stdlib.Effect.t += E : unit Effect.t
 # let f () = perform E;;
 val f : unit -> unit = <fun>
 # f ();;
-Exception: Unhandled.
-# match f () with (* alternative syntax for effect handlers *)
-  | () -> "done"                              (* value case *)
-  | effect E k -> continue k ();;            (* effect case *)
-- : string = "done"
+Exception: Stdlib.Effect.Unhandled(E)
+# open Effect.Deep;;
+# try_with f () {effc = (fun (type c) (eff: c Effect.t) ->
+      match eff with
+      | E -> Some (fun (k: (c,_) continuation) -> continue k ())
+      | _ -> None
+  )};;
+- : unit = ()
 ```
 
-An effect system for OCaml 
-[is in development](https://www.youtube.com/watch?t=2035&v=z8SI7WBtlcA&feature=youtu.be). 
-When effect handlers land in upstream OCaml, they will be equipped with an effect system.
-
 ### Exercise 1: Implement exceptions from effects ★☆☆☆☆
 
 As mentioned before, effects generalise exceptions. Exceptions handlers are
 effect handlers that ignore the continuation. Your task is to implement
 exceptions in terms of effects. The source file is `sources/exceptions.ml`.
 
-## 2. Effectful Computations in a Pure Setting
+## 2. Shallow vs Deep Handlers
+
+The OCaml standard library provides two different modules for handling effects: `Effect.Deep` and `Effect.Shallow`. When a deep handler returns a continuation, the continuation also includes the handler. This means that, when the continuation is resumed, the effect handler is automatically re-installed, and will handle the effect(s) that the computation may perform in the future.
```

Comment:
I wish we hadn't named the module `Effect.Shallow` as the handler therein are **not** shallow. Genuine shallow handlers are more expressive in a certain sense than the (sheep) handlers exposed by that module. I guess it is too late to make this change upstream @kayceesrk.

Reply:
Naming things is hard. Would you call these `Sheep` handlers rather than `Shallow` handlers? In any case, it may be useful to create an issue on ocaml/ocaml with a justification for why `Shallow` handlers may not be the most appropriate name.

Reply:
Can someone link me something on the difference between shallow and sheep handlers? I had trouble finding any information on this